packages/salesforce-adapter/src/filters/centralize_tracking.ts

```diff
@@ -0,0 +1,241 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import _ from 'lodash'
+import { collections, values as valueUtils } from '@salto-io/lowerdash'
+import {
+  Change,
+  Field, getAllChangeData,
+  getChangeData,
+  isAdditionOrModificationChange,
+  isField,
+  isFieldChange,
+  isModificationChange,
+  isObjectType,
+  isObjectTypeChange,
+  ModificationChange,
+  ObjectType,
+  ReferenceExpression,
+  toChange,
+} from '@salto-io/adapter-api'
+import { LocalFilterCreator } from '../filter'
+import { apiName, isCustomObject, isFieldOfCustomObject } from '../transformers/transformer'
+import {
+  FEED_HISTORY_TRACKED_FIELDS,
+  FIELD_ANNOTATIONS,
+  HISTORY_TRACKED_FIELDS,
+  OBJECT_FEED_HISTORY_TRACKING_ENABLED,
+  OBJECT_HISTORY_TRACKING_ENABLED,
+} from '../constants'
+
+const { awu } = collections.asynciterable
+
+type trackedFieldsDefinition = {
+  objectLevelEnable: string
+  fieldLevelEnable: string
+  aggregate: string
+}
+
+const trackedFields: trackedFieldsDefinition[] = [
+  {
+    objectLevelEnable: OBJECT_HISTORY_TRACKING_ENABLED,
+    fieldLevelEnable: FIELD_ANNOTATIONS.TRACK_HISTORY,
+    aggregate: HISTORY_TRACKED_FIELDS,
+  },
+  {
+    objectLevelEnable: OBJECT_FEED_HISTORY_TRACKING_ENABLED,
+    fieldLevelEnable: FIELD_ANNOTATIONS.TRACK_FEED_HISTORY,
+    aggregate: FEED_HISTORY_TRACKED_FIELDS,
+  },
+]
+
+
+const isTrackingEnabled = (type: ObjectType, trackingDef: trackedFieldsDefinition): boolean => (
+  (type.annotations[trackingDef.objectLevelEnable] === true)
+  || (type.annotations[trackingDef.aggregate] !== undefined)
+)
+
+const centralizeTrackingAnnotations = (customObject: ObjectType, trackingDef: trackedFieldsDefinition): void => {
+  if (!isTrackingEnabled(customObject, trackingDef)) {
+    delete customObject.annotations[trackingDef.objectLevelEnable]
+    return
+  }
+
+  customObject.annotations[trackingDef.aggregate] = _(customObject.fields)
+    .pickBy(field => (field.annotations[trackingDef.fieldLevelEnable] === true))
+    .mapValues(field => (field !== undefined ? new ReferenceExpression(field.elemID) : undefined))
+    .value()
+
+  Object.values(customObject.fields).forEach(field => delete field.annotations[trackingDef.fieldLevelEnable])
+  delete customObject.annotations[trackingDef.objectLevelEnable]
+}
+
+const createFieldChangeIfApplicable = async (
```

Comment:
```suggestion
const createFieldChange = async (
```
Nit: The `IfApplicable` is redundant as it's clear we won't always be able to create the `Field Change` since the return type is possibly undefined.

Reply:
Sorry, I made this change to the history tracking PR, but it didn't carry over to this one (presumably because the file was renamed).